### PR TITLE
RA-1968: Add all patient encounters to context model

### PR DESCRIPTION
**Pull request by brandones**
https://issues.openmrs.org/browse/RA-1968

This allows extension `require`s to refer to a patient's historical encounters. Demonstrated here with "Covid 19 Note d'Admission"

previous-encounter

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/clinicianfacing/VisitsSectionFragmentController.java [hunk 1]
```diff
@@ -139,8 +139,11 @@ public void controller(FragmentConfiguration config,
 		}
 
 		Map<VisitDomainWrapper, String> recentVisitsWithLinks = new LinkedHashMap<VisitDomainWrapper, String>();
+
+		if (!recentVisits.isEmpty()) {
+			contextModel.put("visit", new VisitContextModel(recentVisits.get(recentVisits.size() - 1)));
+		}
```
Comment (mseaton) [inline omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/clinicianfacing/VisitsSectionFragmentController.java:145]: Not sure why this change was necessary?  Not really a problem, but I'm curious what this achieved.

**Review by mseaton — APPROVED** [review]
LGTM, one minor comment

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/clinicianfacing/VisitsSectionFragmentController.java (hunk 1 above)
Comment (brandones) [reply omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/clinicianfacing/VisitsSectionFragmentController.java:145]: Just made the code more readable, more plainly expressive of what's actually happening.
